framework/Optimizers/GradientDescent.py

```diff
@@ -565,6 +564,7 @@ def _checkAcceptability(self, traj, opt, optVal, info):
       @ In, info, dict, identifying information about the opt point
       @ Out, acceptable, str, acceptability condition for point
       @ Out, old, dict, old opt point
+      @ Out, rejectReason, int, reject reason of opt point
```

Comment:
please create an enumerator in the base class for this... integers are dangerous because any opt can create its nomenclature or meaning for them...

Reply:
Indeed what are the options?